### PR TITLE
Allow uploading directly to S3

### PR DESCRIPTION
**Pull request by jcmorrow**
It's pretty easy to overwhelm our servers by sending them a ton of CI recordings at once. It's been happening multiple times a day sometimes for weeks. In the case of dozens or hundreds of recordings coming from CI, there is not really any advantage to using a websocket over HTTP, and HTTP is a *lot* simpler (and we don't have to worry about scale ever, it just goes straight to AWS).

### CODE REVIEW

File: packages/replay/src/upload.ts [hunk 1]
```diff
@@ -141,57 +140,14 @@ class ReplayClient {
...
-    }
-
-    debug("%s: Uploaded %d bytes", recordingId, offset);
-
-    // Explicitly mark the recording complete so the server knows that it has
-    // been sent all of the recording data, and can save the recording.
-    // This means if someone presses Ctrl+C, the server doesn't save a
-    // partial recording.
-    await this.client.sendCommand("Internal.finishRecording", { recordingId });
+    const size = fs.statSync(path).size;
```
Comment (loganfsmyth) [inline packages/replay/src/upload.ts:145]: We should avoid the Sync function, especially since we can `await` here with no problem

File: packages/replay/src/main.ts [hunk 2]
```diff
@@ -388,7 +394,7 @@ async function doUploadRecording(
     recordingId,
   });
 
-  await client.connectionUploadRecording(recordingId, recording.path!);
+  await client.connectionUploadRecording(recording.path!, uploadLink);
```
Comment (loganfsmyth) [inline packages/replay/src/main.ts:397]: Looks like we can pass the size in here too?

File: packages/replay/src/main.ts [hunk 3]
```diff
@@ -360,6 +360,8 @@ async function doUploadRecording(
     return recording.id;
   }
 
+  const size = fs.statSync(recording.path!).size;
```
Comment (loganfsmyth) [inline packages/replay/src/main.ts:363]: Let's avoid the `Sync`-suffix functions

File: packages/sourcemap-upload/package-lock.json [hunk 4]
```diff
@@ -6,7 +6,7 @@
 	"packages": {
 		"": {
 			"name": "@replayio/sourcemap-upload",
-			"version": "1.0.6",
+			"version": "1.0.7",
 			"license": "BSD-3-Clause",
```
Comment (loganfsmyth) [inline packages/sourcemap-upload/package-lock.json:10]: Is all this from an unrelated PR? Can we land that first or in a separate commit so it's easier to see what's changing I this PR?

File: packages/replay/src/upload.ts [hunk 5]
```diff
@@ -141,57 +140,14 @@ class ReplayClient {
...
-      await send(buffer);
-    }
-
-    debug("%s: Uploaded %d bytes", recordingId, offset);
-
-    // Explicitly mark the recording complete so the server knows that it has
-    // been sent all of the recording data, and can save the recording.
-    // This means if someone presses Ctrl+C, the server doesn't save a
-    // partial recording.
-    await this.client.sendCommand("Internal.finishRecording", { recordingId });
```
Comment (loganfsmyth) [inline packages/replay/src/upload.ts:194]: We maaaye still want to call this afterward so the API knows when the CLI thinks it is done? At least then if the user disconnects without finishing uploading to S3, we'll be able to see the disconnect-without-finish and clear the DB the DB entry?

Comment (jcmorrow) [reply packages/replay/src/upload.ts:194]: Yeah I think I'm good with that.

File: packages/sourcemap-upload/package-lock.json (hunk 4 above)
Comment (jcmorrow) [reply packages/sourcemap-upload/package-lock.json:10]: I think this was all from me installing `node-fetch`, but maybe I ran some unnecessary commands alongside that. I can try and reduce the noise here.

File: packages/replay/src/upload.ts (hunk 5 above)
Comment (jcmorrow) [reply packages/replay/src/upload.ts:194]: Done!

File: packages/sourcemap-upload/package-lock.json (hunk 4 above)
Comment (jcmorrow) [reply packages/sourcemap-upload/package-lock.json]: Should be significantly reduced now!

**Review by loganfsmyth — APPROVED** [review]
